misk-admin/src/main/kotlin/misk/web/metadata/jvm/JvmMetadataAction.kt

```diff
@@ -0,0 +1,91 @@
+package misk.web.metadata.jvm
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import misk.moshi.adapter
+import misk.web.Get
+import misk.web.RequestContentType
+import misk.web.ResponseContentType
+import misk.web.actions.WebAction
+import misk.web.dashboard.AdminDashboardAccess
+import misk.web.mediatype.MediaTypes
+import java.lang.management.RuntimeMXBean
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Conveys information about the current JVM
+ */
+@Singleton
+class JvmMetadataAction : WebAction {
+
+  private val runtimeMxBean: RuntimeMXBean
+  private val moshiAdapter: JsonAdapter<JvmRuntimeResponse>
+
+  @Inject constructor(runtimeMxBean: RuntimeMXBean, moshi: Moshi) {
+    this.runtimeMxBean = runtimeMxBean
+    // Indent to make the output more readable since the response can be quite large!
+    this.moshiAdapter = moshi.adapter<JvmRuntimeResponse>().indent("  ")
+  }
+
+  @Get("/api/config/jvm/runtime")
+  @RequestContentType(MediaTypes.APPLICATION_JSON)
+  @ResponseContentType(MediaTypes.APPLICATION_JSON)
+  @AdminDashboardAccess
+  fun getRuntime(): String {
+    return moshiAdapter.toJson(JvmRuntimeResponse.create(runtimeMxBean))
+  }
+
+  /**
+   * A data class for serializing information from [java.lang.management.RuntimeMxBean]
+   */
+  data class JvmRuntimeResponse(
+    val pid: Long?,
+    val name: String?,
+    val vm_name: String?,
+    val vm_vendor: String?,
+    val vm_version: String?,
+    val spec_name: String?,
+    val spec_vendor: String?,
+    val spec_version: String?,
+    val management_spec_version: String?,
+    val class_path: String?,
+    val library_path: String?,
+    val is_boot_class_path_supported: Boolean?,
+    val boot_class_path: String?,
+    val input_arguments: List<String>?,
+    val uptime_millis: Long?,
+    val start_time_millis: Long?,
+    val system_properties: Map<String, String>?,
+  ) {
+    companion object Factory {
```

Comment:
I see companion objects left anonymous most of the time, and [the docs](https://kotlinlang.org/docs/object-declarations.html#companion-objects) say the name is optional and not used when referencing the contained static methods. Just curious about your approach:

Whats the motivation behind naming the companion? Would you always name it? 
If a future use case appears for another companion/static method that doesn't fit into a "factory", how would you approach it?

Reply:
My Java is showing through here: I'd normally just make an inner static factory class, but I'll remove the name here as it's not necessary or conventional